Common++/header/IpAddress.h

```diff
@@ -447,6 +463,108 @@ namespace pcpp
+		 * std::invalid_argument exception is thrown
+		 */
+		IPv4Network(const IPv4Address& address, uint8_t prefixLen);
+
+		/**
+		 * A constructor that creates an instance of the class out of an address representing the network prefix
+		 * and a subnet mask
+		 * @param address An address representing the network prefix. If the address is invalid std::invalid_argument
+		 * exception is thrown
+		 * @param subnetMask A string representing a subnet mask in the format of X.X.X.X, for example: 255.255.0.0.
+		 * Please notice that subnets that start with zeros are invalid, for example: 0.0.255.255. The only subnet mask
+		 * starting with zeros that is valid is 0.0.0.0. If the subnet is invalid std::invalid_argument
+		 * exception is thrown
+		 */
+		IPv4Network(const IPv4Address& address, const std::string& subnetMask);
+
+		/**
+		 * A constructor that creates an instance of the class out of a string representing the network prefix and
+		 * a prefix length or a subnet mask
+		 * @param addressAndSubnet A string in one of these formats:
+		 *  - X.X.X.X/Y where X.X.X.X is a valid IPv4 address representing the network prefix and Y is a number between
+		 *    0 and 32 representing the network prefix
+		 *  - X.X.X.X/Y.Y.Y.Y where X.X.X.X is a valid IPv4 address representing the network prefix and Y.Y.Y.Y is
+		 *    a valid subnet mask
+		 *  For any invalid value std::invalid_argument is thrown
+		 */
+		IPv4Network(const std::string& addressAndSubnet);
+
+		/**
+		 * @return The prefix length, for example: the prefix length of 10.10.10.10/255.0.0.0 is 8
+		 */
+		uint8_t getPrefixLen() const;
+
+		/**
+		 * @return The subnet mask, for example: the subnet mask of 10.10.10.10/8 is 255.0.0.0
+		 */
+		std::string getSubnetMask() const { return IPv4Address(m_Mask).toString(); }
+
+		/**
+		 * @return The network prefix, for example: the network prefix of 10.10.10.10/16 is 10.10.0.0
+		*/
+		IPv4Address getNetworkPrefix() const { return IPv4Address(m_NetworkPrefix); }
+
+		/**
+		 * @return The lowest IPv4 address in this network, for example: the lowest address in 10.10.10.10/16 is
+		 * 10.10.0.0
+		 */
+		IPv4Address getLowestAddress() const;
+
+		/**
+		 * @return The highest IPv4 address in this network, for example: the highest address in 10.10.10.10/16 is
+		 * 10.10.255.255
+		 */
+		IPv4Address getHighestAddress() const;
+
+		/**
+		 * @return The number of addresses in this network, for example: the number of addresses in 10.10.0.0/24 is 256
+		 */
+		uint64_t getTotalAddressCount() const;
+
+		/**
+		 * @param address An IPv4 address
+		 * @return True is the address belongs to the network, false otherwise or if the address isn't valid
+		 */
+		bool includes(const IPv4Address& address) const;
+
+		/**
+		 * @param network An IPv4 network
+		 * @return True is the input network is included within this network, false otherwise, for example:
```

Comment:
I would explicit if the network should be totally included or should overlap.

True is the input network is **totally/completely** included within this network